doc/devel/contributing.rst

```diff
@@ -0,0 +1,422 @@
+    <http://www.python.org/dev/peps/pep-0008/>`_. You should consider
+    installing/enabling automatic PEP8 checking in your editor.  Part of the
+    test suite is checking PEP8 compliance, things go smoother if the code is
+    mostly PEP8 compliant to begin with.
+
+  * Each high-level plotting function should have a simple example in
+    the `Example` section of the docstring.  This should be as simple as
+    possible to demonstrate the method.  More complex examples should go
+    in the `examples` tree.
+
+  * Changes (both new features and bugfixes) should be tested. See
+    :ref:`testing` for more details.
+
+  * Import the following modules using the standard scipy conventions::
+
+      import numpy as np
+      import numpy.ma as ma
+      import matplotlib as mpl
+      from matplotlib import pyplot as plt
+      import matplotlib.cbook as cbook
+      import matplotlib.collections as mcol
+      import matplotlib.patches as mpatches
+
+  * If your change is a major new feature, add an entry to `What's new`
+    section by adding a new file in `doc/users/whats_new` (see
+    :file:`doc/users/whats_new/README` for more information).
+
+  * If you change the API in a backward-incompatible way, please
+    document it in `doc/api/api_changes`, by adding a new file describing your
+    changes (see :file:`doc/api/api_changes/README` for more information)
+
+  * See below for additional points about
+    :ref:`keyword-argument-processing`, if code in your pull request
+    does that.
+
+In addition, you can check for common programming errors with the following
+tools:
+
+    * Code with a good unittest coverage (at least 70%, better 100%), check
+      with::
+      
+        pip install coverage
+        python tests.py --with-coverage
+
+    * No pyflakes warnings, check with::
+      
+        pip install pyflakes
+        pyflakes path/to/module.py
+
+.. note::
+
+    The current state of the matplotlib code base is not compliant with all
+    of those guidelines, but we expect that enforcing those constraints on all
+    new contributions will get the overall code base quality in the right
+    direction.
+
+
+.. seealso::
+
+  * :ref:`coding_guidelines`
+  * :ref:`testing`
+  * :ref:`documenting-matplotlib`
+
+
+
+.. _new_contributors:
+
+Issues for New Contributors
+---------------------------
+
+New contributors should look for the following tags when looking for issues.
+We strongly recommend that new contributors tackle "new-contributor-friendly"
+issues (easy, well documented issues, that do not require an understanding of
+the different submodules of matplotlib) and "Easy-fix" issues. This helps the
+contributor become familiar with the contribution workflow, and for the core
+devs to become acquainted with the contributor; besides which, we frequently
+underestimate how easy an issue is to solve!
+
+.. topic:: new-contributor-friendly
+
+    A great way to start contributing to matplotlib is to pick an item from
```

Comment:
Can this note be consolidated with the block above?  There is quite a bit of duplication here.